src/monodroid/jni/android-system.cc

```diff
@@ -192,8 +192,8 @@ AndroidSystem::_monodroid__system_property_get (const char *name, char *sp_value
 	//    ../../../jni/android-system.cc(206,10): warning G20816D19: ‘char* strncpy(char*, const char*, size_t)’ specified bound 93 equals destination size [-Wstringop-truncation] [/home/grendel/vc/xamarin/xamarin-android-worktrees/code-quality-improvements/src/monodroid/monodroid.csproj]
 	//    strncpy (sp_value, env_value, sp_value_len);
 	//
-	strncpy (sp_value, env_value, sp_value_len - 1);
-	sp_value[sp_value_len] = '\0';
+	strncpy (sp_value, env_value, sp_value_len - 2);
```

Comment:
Given that we're using `memcpy()` elsewhere, should we just use `memcpy()` *everywhere*?

Reply:
I'd rather keep using the string functions for string types elsewhere. The size calculations here appear to confuse the compiler which warns about code that is actually valid (on targets which have fortified string manipulation functions, e.g. in desktop builds), so using `memcpy` was the easiest way to shut the compiler up here.